## Profitable App Profiles for the App Store and Google Play Markets

### Step 1: Analyzing Mobile App Data

Mail goal of this project is to make an idea about a free app that would be popular for the english speaking auditory depending on the market ratings.
In this project we'll research markets of Google Play and Apple, cleaning all the unnesessary data and finding the most popular app genre.

### Step 2: Opening and Exploring the Data

In this area we'll create functions to get the information from the markets and to comfortably abserve it.
So, we preapare the needed information as links, we read their data and present it in a more readable way.

First, depending on how big our file is we will use pandas to read and to shorten it a little bit. To do this, we'll first: sort our file by popularity so the most popular apps would be up in the file. We'll do that to leave only the most popular versions of an app if it will have duplicates. Next: remove all the versions of an app except for the most popular one.

Also,
in the function bellow and some of the others we will keep track of the time so we'll see how slow or fast our functions work. 

In [1]:
# this function prints headers so it'll be easier to find the right one for next functions


import csv

def get_csv_headers(data):
    with open(data, encoding='UTF-8') as f:
        reader = csv.reader(f)
        print(next(reader))

In [2]:
# here we take a csv file, sort it by popularity, remove the duplicates
# and return a list of lists as a result to be able to work with the other functions:


import pandas as pd
import csv
import time

def read_csv_as_list_remove_duple(data, app_name, rating_name) -> list:
    start = time.time()
    data = pd.read_csv(data)
    data = data.sort_values(by=rating_name, ascending=False)
    data = data.drop_duplicates(subset=app_name, keep='first')
    data_list = data.values.tolist()
    print(time.time()-start)
    return data_list

In [3]:
# these are links to datasets of apps in the stores


GooglePlayCsv = 'Google-Playstore.csv'
AppleStoreCsv = 'appleAppData.csv'

In [4]:
# this function will be used to read the needed stack of information from a list 
# and count its size on different steps of research:


def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
        return len(dataset)

Here are some examples: 

In [5]:
apple_headers = get_csv_headers(AppleStoreCsv)

['App_Id', 'App_Name', 'AppStore_Url', 'Primary_Genre', 'Content_Rating', 'Size_Bytes', 'Required_IOS_Version', 'Released', 'Updated', 'Version', 'Price', 'Currency', 'Free', 'DeveloperId', 'Developer', 'Developer_Url', 'Developer_Website', 'Average_User_Rating', 'Reviews', 'Current_Version_Score', 'Current_Version_Reviews']


In [6]:
# a sneak peek at Apple Store's dataset:


ios_csv = read_csv_as_list_remove_duple(AppleStoreCsv, 'App_Name', 'Reviews')
explore_data(ios_csv, 0, 4, True)

23.412975549697876
['com.google.ios.youtube', 'YouTube: Watch, Listen, Stream', 'https://apps.apple.com/us/app/youtube-watch-listen-stream/id544007664?uo=4', 'Photo & Video', '17+', 261849088.0, '12.0', '2012-09-11T07:23:19Z', '2021-10-10T23:03:29Z', '16.40.3', 0.0, 'USD', True, 281956209, 'Google LLC', 'https://apps.apple.com/us/developer/google-llc/id281956209?uo=4', 'https://www.youtube.com', 4.68757, 22685334, 4.68757, 22685334]


['com.burbn.instagram', 'Instagram', 'https://apps.apple.com/us/app/instagram/id389801252?uo=4', 'Photo & Video', '12+', 193046528.0, '12.0', '2010-10-06T08:12:41Z', '2021-10-13T11:48:53Z', '209.1', 0.0, 'USD', True, 389801255, 'Instagram, Inc.', 'https://apps.apple.com/us/developer/instagram-inc/id389801255?uo=4', 'http://instagram.com/', 4.7468, 21839585, 4.7468, 21839585]


['com.spotify.client', 'Spotify New Music and Podcasts', 'https://apps.apple.com/us/app/spotify-new-music-and-podcasts/id324684580?uo=4', 'Music', '12+', 154742784.0, '12.0', '2011-

1223511

In [7]:
google_headers = get_csv_headers(GooglePlayCsv)

['', 'App Name', 'App Id', 'Category', 'Rating', 'Rating Count', 'Installs', 'Minimum Installs', 'Maximum Installs', 'Free', 'Price', 'Currency', 'Size', 'Minimum Android', 'Developer Id', 'Developer Website', 'Developer Email', 'Released', 'Last Updated', 'Content Rating', 'Privacy Policy', 'Ad Supported', 'In App Purchases', 'Editors Choice', 'Scraped Time']


In [8]:
# and Google Play's: 


google_csv = read_csv_as_list_remove_duple(GooglePlayCsv, 'App Name', 'Rating Count')
explore_data(google_csv, 0, 4, True)

42.632994174957275
[385470, 'WhatsApp Messenger', 'com.whatsapp', 'Communication', 4.0, 138557570.0, '5,000,000,000+', 5000000000.0, 6265637751, True, 0.0, 'USD', 'Varies with device', 'Varies with device', 'WhatsApp LLC', 'http://www.whatsapp.com/', 'android@support.whatsapp.com', 'Oct 18, 2010', 'Jun 09, 2021', 'Everyone', 'http://www.whatsapp.com/legal/#Privacy', False, False, True, '2021-06-16 02:51:24']


[304824, 'Instagram', 'com.instagram.android', 'Social', 3.8, 120206190.0, '1,000,000,000+', 1000000000.0, 3559871277, True, 0.0, 'USD', 'Varies with device', 'Varies with device', 'Instagram', 'http://help.instagram.com/', 'android-support@instagram.com', 'Apr 3, 2012', 'Jun 14, 2021', 'Teen', 'http://instagram.com/legal/privacy/', True, True, True, '2021-06-16 01:33:46']


[2222701, 'Facebook', 'com.facebook.katana', 'Social', 2.3, 117850066.0, '5,000,000,000+', 5000000000.0, 6782619635, True, 0.0, 'USD', 'Varies with device', 'Varies with device', 'Facebook', 'https://www.face

2177944

### Step 3: Deleting Wrong Data

In this section we clean all the unnessesary data, including the following:
 - apps that are somthing but free. Because we want to create a free app;
 - apps that are not in english. So our research would be based only english speaking auditory.

#### Step 3.1: Isolating free apps

In [9]:
# for this function we go by the app info and check the price. The function returns True if the app is free and None if not


def is_free(row, price_index):
    if row[price_index] == 0.0:
        return True

Check on how well everything works so far:

In [10]:
# for Apple Store


import time

clean_ios = []
different = []
start = time.time()
for row in ios_csv:
    if is_free(row, 10):
        clean_ios.append(row)
    else:
        different.append(row)
print(time.time()-start)
explore_data(clean_ios, 1, 5, True)

0.2879974842071533
['com.burbn.instagram', 'Instagram', 'https://apps.apple.com/us/app/instagram/id389801252?uo=4', 'Photo & Video', '12+', 193046528.0, '12.0', '2010-10-06T08:12:41Z', '2021-10-13T11:48:53Z', '209.1', 0.0, 'USD', True, 389801255, 'Instagram, Inc.', 'https://apps.apple.com/us/developer/instagram-inc/id389801255?uo=4', 'http://instagram.com/', 4.7468, 21839585, 4.7468, 21839585]


['com.spotify.client', 'Spotify New Music and Podcasts', 'https://apps.apple.com/us/app/spotify-new-music-and-podcasts/id324684580?uo=4', 'Music', '12+', 154742784.0, '12.0', '2011-07-14T11:22:37Z', '2021-10-08T06:46:57Z', '8.6.68', 0.0, 'USD', True, 324684583, 'Spotify Ltd.', 'https://apps.apple.com/us/developer/spotify-ltd/id324684583?uo=4', 'https://www.spotify.com/', 4.78328, 18893225, 4.78328, 18893225]


['net.kortina.labs.Venmo', 'Venmo', 'https://apps.apple.com/us/app/venmo/id351727428?uo=4', 'Finance', '4+', 372536320.0, '12.0', '2010-04-03T05:41:47Z', '2021-10-08T04:00:00Z', '9.5.1', 

1121732

In [11]:
explore_data(different, 1, 5, True)

['com.WhatsPad.messenger', 'Messaging for WhatsApp on iPad', 'https://apps.apple.com/us/app/messaging-for-whatsapp-on-ipad/id1131873369?uo=4', 'Social Networking', '4+', 5654528.0, '11.0', '2018-07-01T06:37:30Z', '2019-04-03T10:06:18Z', '1.1.3', 2.99, 'USD', False, 776155467, 'Burak Acemoglu', 'https://apps.apple.com/us/developer/burak-acemoglu/id776155467?uo=4', nan, 4.51323, 299904, 4.51323, 299904]


['com.jackadam.darksky', 'Dark Sky Weather', 'https://apps.apple.com/us/app/dark-sky-weather/id517329357?uo=4', 'Weather', '4+', 44150784.0, '11.0', '2012-04-16T07:00:00Z', '2021-06-22T17:41:23Z', '6.8.6', 3.99, 'USD', False, 284417353, 'Apple', 'https://apps.apple.com/us/developer/apple/id284417353?mt=12&uo=4', 'https://darksky.net/app', 4.70354, 276271, 4.70354, 276271]


['com.novosoft.TurboScan', 'TurboScan™ Pro: PDF scanner', 'https://apps.apple.com/us/app/turboscan-pro-pdf-scanner/id342548956?uo=4', 'Business', '4+', 31724544.0, '8.0', '2009-12-07T17:46:12Z', '2021-08-05T19:22:44Z

101779

After this step we've got 101779 apps that cost something to 1121731 that we need.

In [12]:
# for Google Play Store


clean_google = []
different = []
for row in google_csv:
    if is_free(row, 10):
        clean_google.append(row)
    else:
        different.append(row)
explore_data(clean_google, 0, 5, True)

[385470, 'WhatsApp Messenger', 'com.whatsapp', 'Communication', 4.0, 138557570.0, '5,000,000,000+', 5000000000.0, 6265637751, True, 0.0, 'USD', 'Varies with device', 'Varies with device', 'WhatsApp LLC', 'http://www.whatsapp.com/', 'android@support.whatsapp.com', 'Oct 18, 2010', 'Jun 09, 2021', 'Everyone', 'http://www.whatsapp.com/legal/#Privacy', False, False, True, '2021-06-16 02:51:24']


[304824, 'Instagram', 'com.instagram.android', 'Social', 3.8, 120206190.0, '1,000,000,000+', 1000000000.0, 3559871277, True, 0.0, 'USD', 'Varies with device', 'Varies with device', 'Instagram', 'http://help.instagram.com/', 'android-support@instagram.com', 'Apr 3, 2012', 'Jun 14, 2021', 'Teen', 'http://instagram.com/legal/privacy/', True, True, True, '2021-06-16 01:33:46']


[2222701, 'Facebook', 'com.facebook.katana', 'Social', 2.3, 117850066.0, '5,000,000,000+', 5000000000.0, 6782619635, True, 0.0, 'USD', 'Varies with device', 'Varies with device', 'Facebook', 'https://www.facebook.com/facebook',

2134388

In [13]:
explore_data(different, 0, 5, True)

[837351, 'Minecraft', 'com.mojang.minecraftpe', 'Arcade', 4.6, 4132618.0, '10,000,000+', 10000000.0, 36183322, False, 7.49, 'USD', 'Varies with device', '5.0 and up', 'Mojang', 'http://help.mojang.com', 'help@minecraft.net', 'Aug 15, 2011', 'May 27, 2021', 'Everyone 10+', 'https://privacy.microsoft.com/en-us/privacystatement', False, True, True, '2021-06-16 09:47:13']


[508696, 'Hitman Sniper', 'com.squareenixmontreal.hitmansniperandroid', 'Action', 4.4, 854992.0, '10,000,000+', 10000000.0, 24958893, False, 0.99, 'USD', '48M', '4.1 and up', 'SQUARE ENIX LTD', 'http://www.square-enix.com/eng/', 'support@square-enix-montreal.mail.helpshift.com', 'Jun 3, 2015', 'Aug 24, 2020', 'Mature 17+', 'http://eu.square-enix.com/en/documents/privacy-index', True, True, True, '2021-06-16 04:42:58']


[1093849, 'Geometry Dash', 'com.robtopx.geometryjump', 'Arcade', 4.6, 780332.0, '1,000,000+', 1000000.0, 4827946, False, 1.99, 'USD', '85M', '4.0 and up', 'RobTop Games', 'http://www.robtopgames.com', 's

43556

Google statistic for free apps is quite surprising in comparison: 43556 expensive apps to 2134388 free

#### Step 3.2: Leaving english apps only

On this step we would leave only those apps which are avaliable in English by checking on an app having foreign symbols in its symbols.
Each symbol in ascii is coded with a number less than or equal to 127. And by comparing each of the letters to this number we'll get whether it is ascii or not.

Also, 
you see, there is one problem that appears in this code below:

In [14]:
print(ios_csv[45])
print(ord('®'))

['com.citigroup.citimobile', 'Citi Mobile®', 'https://apps.apple.com/us/app/citi-mobile/id301724680?uo=4', 'Finance', '4+', 236280832.0, '12.3', '2009-01-24T08:00:00Z', '2021-09-17T19:15:53Z', '9.44.0', 0.0, 'USD', True, 301724683, 'Citibank', 'https://apps.apple.com/us/developer/citibank/id301724683?uo=4', 'http://www.citibank.com', 4.86892, 2588063, 4.86892, 2588063]
174


We get a number(128540) much heigher than 127 through the app's name is fully in english('Instachat 😜').
To fight this problem we add a count of non ascii symbols inside of the name and only if the count is highter than 3 we take the app away.

In [15]:
# main function is based on identifing non ascii characters using an ord() function


def is_english(row, name_index):
    non_ASCII = 0
    try:
        for letter in row[name_index]:
            if ord(letter) > 127:
                non_ASCII += 1
    except TypeError:
        return False
    if non_ASCII > 3:
        return False
    else:
        return True

Function testing:

In [16]:
import time

start = time.time()

clean = []
different = []
for row in ios_csv:
    if is_english(row, 1):
        clean.append(row)
    else:
        different.append(row)
english_apps = explore_data(clean, 0, 4, True)
non_english_apps = explore_data(different, 0, 4, True)
print(time.time()-start)

['com.google.ios.youtube', 'YouTube: Watch, Listen, Stream', 'https://apps.apple.com/us/app/youtube-watch-listen-stream/id544007664?uo=4', 'Photo & Video', '17+', 261849088.0, '12.0', '2012-09-11T07:23:19Z', '2021-10-10T23:03:29Z', '16.40.3', 0.0, 'USD', True, 281956209, 'Google LLC', 'https://apps.apple.com/us/developer/google-llc/id281956209?uo=4', 'https://www.youtube.com', 4.68757, 22685334, 4.68757, 22685334]


['com.burbn.instagram', 'Instagram', 'https://apps.apple.com/us/app/instagram/id389801252?uo=4', 'Photo & Video', '12+', 193046528.0, '12.0', '2010-10-06T08:12:41Z', '2021-10-13T11:48:53Z', '209.1', 0.0, 'USD', True, 389801255, 'Instagram, Inc.', 'https://apps.apple.com/us/developer/instagram-inc/id389801255?uo=4', 'http://instagram.com/', 4.7468, 21839585, 4.7468, 21839585]


['com.spotify.client', 'Spotify New Music and Podcasts', 'https://apps.apple.com/us/app/spotify-new-music-and-podcasts/id324684580?uo=4', 'Music', '12+', 154742784.0, '12.0', '2011-07-14T11:22:37Z', '

In [17]:
import time

start = time.time()

clean = []
different = []
for row in google_csv:
    if is_english(row, 1):
        clean.append(row)
    else:
        different.append(row)
english_apps = explore_data(clean, 0, 4, True)
non_english_apps = explore_data(different, 0, 4, True)
print(time.time()-start)

[385470, 'WhatsApp Messenger', 'com.whatsapp', 'Communication', 4.0, 138557570.0, '5,000,000,000+', 5000000000.0, 6265637751, True, 0.0, 'USD', 'Varies with device', 'Varies with device', 'WhatsApp LLC', 'http://www.whatsapp.com/', 'android@support.whatsapp.com', 'Oct 18, 2010', 'Jun 09, 2021', 'Everyone', 'http://www.whatsapp.com/legal/#Privacy', False, False, True, '2021-06-16 02:51:24']


[304824, 'Instagram', 'com.instagram.android', 'Social', 3.8, 120206190.0, '1,000,000,000+', 1000000000.0, 3559871277, True, 0.0, 'USD', 'Varies with device', 'Varies with device', 'Instagram', 'http://help.instagram.com/', 'android-support@instagram.com', 'Apr 3, 2012', 'Jun 14, 2021', 'Teen', 'http://instagram.com/legal/privacy/', True, True, True, '2021-06-16 01:33:46']


[2222701, 'Facebook', 'com.facebook.katana', 'Social', 2.3, 117850066.0, '5,000,000,000+', 5000000000.0, 6782619635, True, 0.0, 'USD', 'Varies with device', 'Varies with device', 'Facebook', 'https://www.facebook.com/facebook',

The statistic shows that in 2021 in Google Play there are 7 times more foreign apps than on Apple Store. By the way the situation in 2018 was turned upside down!

#### Step 3.3: Filling nan values

During our research we found that there are some so called 'nan' values that appear in the coulumn of installs, which is very valuable as the creteria for popularity that we will use later. All together there just 107 apps with nan value of 'Installs' for Google Play Store and 0 for Apple Store in 'Reviews'

In [18]:
count = 0

for app in ios_csv:           
    n_installs = app[-3]
    if type(n_installs) == float:
        count += 1
        continue

print(count)

0


In [19]:
# an example of row with a nan value:


print(google_csv[2174036])

[1891126, 'Gıybettin', 'onurgule.giybettin', 'Social', nan, nan, nan, nan, 5756, False, 0.0, nan, '24M', '5.0 and up', 'Onurgule', 'https://giybettin.com', 'info@giybettin.com', 'Jul 22, 2019', 'Apr 11, 2020', 'Everyone', 'https://giybettin.com/privacy.pdf', True, False, False, '2021-06-16 06:50:56']


In [20]:
x = float('nan')
y = 5
print(x+y)
print(x*y)
print(x/y)
print(x-y)

nan
nan
nan
nan


About nan values: nan is an acronym for 'Not a Number'. It represents an undefined or missing value and has a type float which makes it imposible to make any data related mathematical operations with it because any result will be a nan. 

For handeling the nan cases we will redefine them to 0 using iteration. 

In [21]:
import math

def filling_nan(row):
    for i, value in enumerate(row):
        if type(value) is float:
            if math.isnan(value):
                row[i] = 0
    return row

In [22]:
# quick check:


import math

clean = []
for row in google_csv:
    clean.append(filling_nan(row))
    
count = 0
for row in google_csv:
    for value in row:
        if type(value) == float:
            if math.isnan(value):
                count += 1

print(count)

0


#### Step 3.4: Main 

Here we will combine all the previous functions into one big. First to make the prosess universal for both google play and apple stores we will create a function that helps us find the right index of position by its name

In [23]:
import time

def data_cleaning(data, price_index, name_index):
    start = time.time()
    result = []
    for row in data:
        new_row = filling_nan(row)
        if is_free(new_row, price_index) and is_english(new_row, name_index):
            result.append(new_row)
    print(time.time()-start)
    return result

Our final cleaned datasets are ready. Take a look on them:

In [24]:
google_clean = data_cleaning(google_csv, price_index=10, name_index=1)

explore_data(google_clean, 0, 5, True)

11.686027765274048
[385470, 'WhatsApp Messenger', 'com.whatsapp', 'Communication', 4.0, 138557570.0, '5,000,000,000+', 5000000000.0, 6265637751, True, 0.0, 'USD', 'Varies with device', 'Varies with device', 'WhatsApp LLC', 'http://www.whatsapp.com/', 'android@support.whatsapp.com', 'Oct 18, 2010', 'Jun 09, 2021', 'Everyone', 'http://www.whatsapp.com/legal/#Privacy', False, False, True, '2021-06-16 02:51:24']


[304824, 'Instagram', 'com.instagram.android', 'Social', 3.8, 120206190.0, '1,000,000,000+', 1000000000.0, 3559871277, True, 0.0, 'USD', 'Varies with device', 'Varies with device', 'Instagram', 'http://help.instagram.com/', 'android-support@instagram.com', 'Apr 3, 2012', 'Jun 14, 2021', 'Teen', 'http://instagram.com/legal/privacy/', True, True, True, '2021-06-16 01:33:46']


[2222701, 'Facebook', 'com.facebook.katana', 'Social', 2.3, 117850066.0, '5,000,000,000+', 5000000000.0, 6782619635, True, 0.0, 'USD', 'Varies with device', 'Varies with device', 'Facebook', 'https://www.face

1954372

In [25]:
ios_clean = data_cleaning(ios_csv, price_index=10, name_index=1)

explore_data(ios_clean, 0, 5, True)

5.052011966705322
['com.google.ios.youtube', 'YouTube: Watch, Listen, Stream', 'https://apps.apple.com/us/app/youtube-watch-listen-stream/id544007664?uo=4', 'Photo & Video', '17+', 261849088.0, '12.0', '2012-09-11T07:23:19Z', '2021-10-10T23:03:29Z', '16.40.3', 0.0, 'USD', True, 281956209, 'Google LLC', 'https://apps.apple.com/us/developer/google-llc/id281956209?uo=4', 'https://www.youtube.com', 4.68757, 22685334, 4.68757, 22685334]


['com.burbn.instagram', 'Instagram', 'https://apps.apple.com/us/app/instagram/id389801252?uo=4', 'Photo & Video', '12+', 193046528.0, '12.0', '2010-10-06T08:12:41Z', '2021-10-13T11:48:53Z', '209.1', 0.0, 'USD', True, 389801255, 'Instagram, Inc.', 'https://apps.apple.com/us/developer/instagram-inc/id389801255?uo=4', 'http://instagram.com/', 4.7468, 21839585, 4.7468, 21839585]


['com.spotify.client', 'Spotify New Music and Podcasts', 'https://apps.apple.com/us/app/spotify-new-music-and-podcasts/id324684580?uo=4', 'Music', '12+', 154742784.0, '12.0', '2011-0

1099863

So in the end of validation we are left with 1'954'372 apps in Google Play Store and 1'099'377 apps in Apple Store

### Step 4: Most Common Apps by Genre

Our main goal is to figure it out what kind of app will more likely to attract a lot of attention. So far we have cleaned the data for the research and now we would look at the most common genres for creating an app.
 Here we'll make a better view of our data:

In [26]:
# creating a dictionary with [name_of_a_genre] = percentage of apps MADE in this genre:


def freq_table(data, index):
    percent_100 = 0
    freq = {}
    for row in data:
        percent_100 += 1
        criteria = row[index]
        if criteria in freq.keys():
            freq[criteria] += 1
        else: 
            freq[criteria] = 1
    for key, value in freq.items():
        freq[key] = round(value * 100 / percent_100, 3)
    return freq

In [27]:
# sorting dictionary & creating a better view on the picture:


def display_table(data, index):
    table = freq_table(data, index)
    table_sorted = sorted(table.items(), key=lambda item: item[1], reverse=True)
    for entry in table_sorted:
        print(entry[0], ':', entry[1])

__The most common apps by genre for Gooogle Play Store:__

In [ ]:
display_table(google_clean, 3)

If you'll look at this you might think that the best desision is to build an educational app because it's the most common category

__Same for Apple Store:__

In [ ]:
display_table(ios_clean, 3)

For Apple Store situation is different. If you look at the first most comon category you will see 'Games' which was not at the first positions in Google Play as well. Maybe because Google Play developers decided to devide one category into many small ones.

Education still has 9 persent of all apps as in Google Play Store.

### Step 5: Most Popular Apps by Genre

That's the final step of our observing. We'll look at 'Category' and 'Installs' in Google Play; and 'Primary_Genre' and 'Reviews' in Apple Store. The abserving will include two different cycles of building raitings:
 - In the first we'll go by genres
 - In the second - by apps.

#### Step 5.1: Apple Store

In [ ]:
# main function for ios

genre_index = 3
review_index = -3
unique_genres = freq_table(ios_clean, genre_index).keys()
result = {}
for genre in unique_genres:
    total = 0
    len_genre = 0
    for app in ios_clean:
        genre_app = app[genre_index]
        if genre_app == genre:
            total += float(app[review_index])
            len_genre += 1
    avg_ratings = total / len_genre
    result[genre] = avg_ratings
table_sorted = sorted(result.items(), key=lambda item: item[1], reverse=True)
for entry in table_sorted:
    print(entry[0], ':', entry[1])

The most popular category's place found in its strange ways Weather(3671) with a small gap from the second place - Photo & Video with 3666.

Games at the same point took the 6th place.

#### Step 5.2: Google Play Store

In [ ]:
# main function for android
genre_index = 3
review_index = 6
unique_genres = freq_table(google_clean, genre_index).keys()

for category in unique_genres:
    total = 0
    len_category = 0
    for app in google_clean:
        category_app = app[genre_index]
        if category_app == category:            
            n_installs = app[review_index]
            if type(n_installs) == int:
                total += n_installs
                continue
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

COMMUNICATION is the most popular category for Google Play. Which is not surprising with the giants like 'WhatsApp' and 'Google Chrome'. Depending on the pretty small diversity of this genre that's not the best idea for an area to make an app in.

In [ ]:
for app in google_clean:
    if app[3] == 'Communication' and (app[6] == '5,000,000,000+'
                                      or app[6] == '1,000,000,000+'
                                      or app[6] == '500,000,000+'):
        print(app[1], ':', app[6])

In [ ]:
for app in google_clean:
    if app[3] == 'Board' and (app[6] == '5,000,000,000+'
                                      or app[6] == '1,000,000,000+'
                                      or app[6] == '500,000,000+'):
        print(app[1], ':', app[6])

In [ ]:
for app in google_clean:
    if app[3] == 'Board' and (app[6] == '100,000,000+'
                                      or app[6] == '50,000,000+'
                                      or app[6] == '10,000,000+'):
        print(app[1], ':', app[6])

I suggest to make an app in the Board genre. It is middle by popularity and the apps in this category are mostly the classic board games like Roll-and-move or Colour. So they hold to the classical disigns. I think it will be easier to highlight yourself in this category by creating a more complex and absurd game. 

### Conclucion

In this project our goal was to make a recomendation of an app profile that would be popular on the Google Play Market. 

During this prosses we were organazing data, validating and researching it. We used such libraries as: math, time, pandas, csv. 

Aside from the main research we learned about nan values and sistem of ASCII encoding.